pkg/kubectl/resource/builder.go

```diff
@@ -224,6 +224,28 @@ func (b *Builder) ResourceTypeOrNameArgs(args ...string) *Builder {
 	return b
 }
 
+// SelectedResourceTypeOrNameArgs indicates that the build should accept one or two arguments
+// of the form `(<type1>[,<type2>,...]|<type> <name>)`. When one argument is received, the types
+// retrieved will be retrieved from the server. The resources retrieved must be restricted via
+// a label selector othrerwise an error is reported. If more that two arguments are provided an
+// error is set.
+func (b *Builder) SelectedResourceTypeOrNameArgs(args ...string) *Builder {
```

Comment:
I think the name is slightly confusing - it wasn't clear to me the distinction at first.  Maybe we should add a boolean argument to ResourceTypeOrNameArgs "allowEmptySelector"?